src/VehicleSelect/VehicleSelect.test.tsx

```diff
@@ -0,0 +1,165 @@
+      modelYear: "2015",
+      projectCode: "911",
+      variant: "MP"
+    },
+    {
+      _id: "64c8c4cccc8d6f00130b367e",
+      modelYear: "2015",
+      projectCode: "911",
+      variant: "JS"
+    },
+    {
+      _id: "64c8c4cccc8d6f00130b3691",
+      modelYear: "2016",
+      projectCode: "911",
+      variant: "DB"
+    },
+    {
+      _id: "64c8c4cccc8d6f00130b36a4",
+      modelYear: "2016",
+      projectCode: "911",
+      variant: "MC"
+    }
+  ],
+  onVehicleChange: () => {},
+  selectedVehicles: []
+};
+
+/**
+ * Test wrapper for VehicleSelect
+ */
+
+const VehicleSelectWithState = ({
+  onVehicleChange,
+  selectedVehicles: valueIn = [],
+  ...rest
+}: VehicleSelectProps) => {
+  const [value, setValue] = React.useState(valueIn);
+
+  const handleChange = selectedValues => {
+    setValue(selectedValues);
+    onVehicleChange(selectedValues);
+  };
+  return (
+    <VehicleSelect
+      {...rest}
+      onVehicleChange={handleChange}
+      selectedVehicles={value}
+    />
+  );
+};
+
+/**
+ * Tests
+ */
+describe("Vehicle Select", () => {
+  it("renders component", () => {
+    render(<VehicleSelectWithState {...defaultProps} />);
+  });
+  it("onVehicleChange called", async () => {
+    const onVehicleChange = jest.fn();
+    render(
+      <VehicleSelectWithState
+        {...defaultProps}
+        onVehicleChange={onVehicleChange}
+      />
+    );
+
+    // open the project selector
+    await userEvent.click(
+      screen.getByRole("combobox", { name: /project code/i })
+    );
+    // click the first option
+    await userEvent.click(screen.getByRole("option", { name: /911/i }));
+
+    // open the model year selector
+    await userEvent.click(
+      screen.getByRole("combobox", { name: /model year/i })
+    );
+    // click the first option
+    await userEvent.click(screen.getByRole("option", { name: /2015/i }));
+
+    // open the vehicle variant selector
+    await userEvent.click(
+      screen.getByRole("combobox", {
+        name: /vehicle variant/i
+      })
+    );
+    // click the first option
+    await userEvent.click(screen.getByRole("option", { name: /JS/i }));
+
+    // open the vehicle variant selector
+    await userEvent.click(
+      screen.getByRole("combobox", {
+        name: /vehicle variant/i
+      })
+    );
+    // click the second option
+    await userEvent.click(screen.getByRole("option", { name: /MP/i }));
+
+    // open the gate selector
+    await userEvent.click(
+      screen.getByRole("combobox", {
+        name: /gate/i
+      })
+    );
+    // click the first option
+    await userEvent.click(screen.getByRole("option", { name: /Gate 1/i }));
+
+    // open the gate selector
+    await userEvent.click(
+      screen.getByRole("combobox", {
+        name: /gate/i
+      })
+    );
+    // click the second option
+    await userEvent.click(screen.getByRole("option", { name: /Gate 2/i }));
+
+    expect(onVehicleChange).toHaveBeenCalledTimes(6);
```

Comment:
Check that the correct value is returned when onVehicleChange is called rather than just checking onVehicleChange has been called.

Reply:
updated test to check what we return on each onVehicleChange